### ISSUE TITLE
`winning_strategy` not set when using `throw :warden`

### ISSUE THREAD

**Issue by f1sherman** [wardencommunity/warden#171]
I noticed as of version 1.2.8 that if `throw :abort` is used from within a strategy, `winning_strategy` is no longer set. This appears to be due to the change in #144.

Should this new behavior be considered expected or a bug?

**Comment by jsmestad** [comment]
@f1sherman mind spiking on a spec to test this against? It appears we overlooked the spec in #144 and we should get a spec together before patching it.

**Comment by f1sherman** [comment]
@jsmestad sure thing! How does this look?
https://github.com/wardencommunity/warden/compare/master...f1sherman:add-spec-for-winning-strategy-not-set-on-throw?expand=1

```
$ bundle exec rspec spec/warden/proxy_spec.rb:646
Run options: include {:locations=>{"./spec/warden/proxy_spec.rb"=>[646]}}

Warden::Proxy
  messages
    should allow access to the failure message (FAILED - 1)

Failures:

  1) Warden::Proxy messages should allow access to the failure message
     Failure/Error: expect(result.last).to eq(["The Fails Strategy Has Failed You"])

       expected: ["The Fails Strategy Has Failed You"]
            got: [nil]

       (compared using ==)

       Diff:
       @@ -1,2 +1,2 @@
       -["The Fails Strategy Has Failed You"]
       +[nil]

     # ./spec/warden/proxy_spec.rb:654:in `block (3 levels) in <top (required)>'

Finished in 0.1097 seconds (files took 0.23948 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/warden/proxy_spec.rb:646 # Warden::Proxy messages should allow access to the failure message
```

**Comment by landongrindheim** [comment]
This issue can be closed now that #175 has been merged
